docs/reference/ml/ml-shared.asciidoc

```diff
@@ -507,6 +507,357 @@ tag::data-frame-analytics-stats[]
 An array of statistics objects for {dfanalytics-jobs}, which are
 sorted by the `id` value in ascending order.
 
+//Begin analysis_stats
+`analysis_stats`::
+(object)
+An object containing statistical data about the analysis.
++
+.Properties of `analysis_stats`
+[%collapsible%open]
+====
+//Begin classification_stats
+`classification_stats`:::
+(object)
+An object containing statistical data about the {classanalysis}.
++
+.Properties of `classification_stats`
+[%collapsible%open]
+=====
+//Begin class_hyperparameters
+`hyperparameters`::::
+(object)
+An object containing the hyperparameters of the {classanalysis}.
++
+.Properties of `hyperparameters`
+[%collapsible%open]
+======
+`class_assignment_objective`::::
+(string)
+Defines whether class assignment maximizes the accuracy or the minimum recall 
+metric. Possible values are `accuracy` and `minimum_recall`.
+
+`downsample_factor`::::
+(double)
+The value of the downsample factor.
+
+`eta`::::
+(double)
+The value of the eta hyperparameter.
+
+`eta_growth_rate_per_tree`::::
+(double)
+Specifies with what rate to increase `eta` for every new tree added to the 
+forest. The rate of 1.0 does not change `eta`, the rate of 1.05 increases `eta` 
+by 5%.
+
+`feature_bag_fraction`::::
+(double)
+The fraction of features that is used when selecting a random bag for each 
+candidate split.
+
+`max_attempts_to_add_tree`::::
+(integer)
+If the algorithm fails to determine a non-trivial tree (more than a single 
+leaf), this parameter determines how many of such consecutive failures are 
+tolerated. Once the number of attempts exceeds the threshold, the forest 
+training stops.
+
+`max_optimization_rounds_per_hyperparameter`::::
+(integer)
+A multiplier responsible for determining the maximum number of 
+hyperparameter optimization steps in the bayesian optimization procedure. 
+Maximum number of steps is determined as the number of undefined hyperparameters 
+times the maximum optimization rounds per hyperparameter.
+
+`max_trees`::::
+(integer)
+The maximum number of trees.
+
+`num_folds`::::
+(integer)
+Maximum number of folds for the cross-validation procedure.
+
+`num_splits_per_feature`::::
+(integer)
+Determines the maximum number of splits for every feature that can occur in a 
+decision tree when the tree is trained.
+
+`regularization_depth_penalty_multiplier`::::
```

Comment:
After https://github.com/elastic/ml-cpp/pull/1096 is through, it will be called `alpha`, it can be cross-referenced with analysis config.